### ISSUE TITLE
NuGet V3 API Support

### ISSUE THREAD

**Issue by rdcm** [GlitchEnzo/NuGetForUnity#229]
Hi!

We trying use NuGetForUnity with our local nuget server (BaGet), but NFU cant fetch packages from local server.

Nuget.config

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
   <packageSources>
         <add key="NuGet" value="http://www.nuget.org/api/v2/" />
         <add key="nuget_host" value="http://nuget_host:5000/v3/" />
   </packageSources>
   <disabledPackageSources />
   <activePackageSource>
      <add key="All" value="(Aggregate source)" />
   </activePackageSource>
   <config>
      <add key="repositoryPath" value="./Packages" />
      <add key="DefaultPushSource" value="http://www.nuget.org/api/v2/" />
   </config>
</configuration>
```

NuGetForUnity - 1.0.3
Unity - 2018.3.5f1

Is there any solution?

**Comment by jwittner** [comment]
Right now N4U only supports the v2 api. Does your server support v2? If not would love contributions to add v3 support. =)

**Comment by rdcm** [comment]
@jwittner  Still only v2 support? :)

**Comment by jwittner** [comment]
@rdcm, unfortunately so. Looking at our use of the nuget API, shouldn't be too complex to support both and could mostly be localized to [NugetPackageSource.cs](https://github.com/GlitchEnzo/NuGetForUnity/blob/c48d1a240bfba21eaf00b3045ae73fa1622456d3/Assets/NuGet/Editor/NugetPackageSource.cs):

* Detect if [ExpandedPath](https://github.com/GlitchEnzo/NuGetForUnity/blob/c48d1a240bfba21eaf00b3045ae73fa1622456d3/Assets/NuGet/Editor/NugetPackageSource.cs#L31) is v2 or v3
* Update API calls to make the right call / parse the right results for the operations:
  * [FindPackagesById](https://github.com/GlitchEnzo/NuGetForUnity/blob/c48d1a240bfba21eaf00b3045ae73fa1622456d3/Assets/NuGet/Editor/NugetPackageSource.cs#L137)
  * [Search](https://github.com/GlitchEnzo/NuGetForUnity/blob/c48d1a240bfba21eaf00b3045ae73fa1622456d3/Assets/NuGet/Editor/NugetPackageSource.cs#L202)
  * [GetUpdates](https://github.com/GlitchEnzo/NuGetForUnity/blob/c48d1a240bfba21eaf00b3045ae73fa1622456d3/Assets/NuGet/Editor/NugetPackageSource.cs#L441)

**Comment by JoC0de** [comment]
fixed in #549
